### DISCUSSION TITLE
Choose a good name for this library

### DISCUSSION THREAD

**Discussion by fzyzcjy** [fzyzcjy/flutter_rust_bridge#72] [Q&A] [answered]
## Context

In addition to what is discussed in this issue, there are also some discussions in Reddit:

https://www.reddit.com/r/rust/comments/q0wrhr/comment/hfbnom7/?utm_source=share&utm_medium=web2x&context=3

## Names for other code generators

Let's look at how others name it.

python

* pyo3 4.7k
* rust-cpython 1.5k
* milksnake 0.7k (inactive)

c/c++

* cxx 3.2k
* cbindgen 1.2k
* autocxx 1.2k
* cc-rs 0.8k
* rust-cpp 0.5k
* flapigen-rs 0.5k (multi language)

java

* flapigen-rs 0.5k (multi language)
* jni-rs 0.5k
* java-rust-example 0.3k
* j4rs 0.2k
* robusta 0.1k
* rjni 0.06k

javascript/js/node

* neon 5.8k
* napi-rs 1.1k
* node-bindgen 0.2k

go/golang

* rust-plus-golang 0.3k
* rure-go 0.2k

## Official doc

https://rust-lang.github.io/api-guidelines/naming.html

> Crate names should not use -rs or -rust as a suffix or prefix. Every crate is Rust! It serves no purpose to remind users of this constantly.

https://github.com/rust-lang/api-guidelines/discussions/29

## Most downloaded crates

https://crates.io/crates?sort=recent-downloads

* cfg-if
* range_core
* rand
* syn
* proc-macro2
* libc
* unicode-xid
* quote
* getrandom
* rand_chacha
* autocfg
* serde
* bitflags
* memchr
* base64
* lazy_static

**Comment by shreyashsaitwal** [comment]
Before you publish the packages to crates.io and pub.dev, do you have any plans to rename the project like suggested by some people on Reddit? To me, `flutter_rust_bridge` seems something very Flutter-related at first glance, which in fact is not true since the package is 100% compatible with pure Dart as well.

**Comment by fzyzcjy** [comment]
@shreyashsaitwal Ah good question! Indeed people in Reddit also suggests:

> If you plan on releasing it on both, I'd give it the same name so people don't get confused. A bunch of crates have rust or rs in them, its not a big deal.

So maybe I should use the *same* name to publish package for both flutter/dart and rust.

> To me flutter_rust_bridge, seems something very Flutter-related at first glance, which in fact is not true since the package is 100% compatible with pure Dart as well.

Yes I totally agree. I originally thought it should be named dart_rust_bridge. But people in Reddit seems to be interested in how it can ease the programming Flutter+Rust. In addition, most people who uses Dart are using Flutter. Dart seems not to be widely used and the main usage is for Flutter.

I mention "Dart" in README as well. Let me emphasize it a bit more.

**Comment by fzyzcjy** [comment]
README edited: https://github.com/fzyzcjy/flutter_rust_bridge/commit/efc14e58ecd7872b2486e3a076bc8d61208c93ea

**Comment by shreyashsaitwal** [comment]
> So maybe I should use the same name to publish package for both flutter/dart and rust.

Yes, I think it makes more sense to publish it under the same name. 

Also, do you have any nominees for the new name?

**Comment by fzyzcjy** [comment]
@shreyashsaitwal Good question. People also suggest `flechette` and `flap`.

IMHO, if someone wants a ffi generator between Flutter/Dart and Rust, he may Google: `flutter rust` or things like that. The package name makes it possible to be found. On the contrary, if it has a fancy name, people may not be able to find it unless this package is really famous. Just my two cents.

**Comment by shreyashsaitwal** [comment]
It doesn't really matter to me what type of name you opt-in for -- fancy or descriptive. It's just that the current name seems too long to me. IMO, lib/package names shouldn't really exceed more than two words.

**Comment by fzyzcjy** [comment]
Good question. What do you think if I create a new Reddit post (or just add a comment in the current post?) asking for people to suggest and vote names?

**Comment by shreyashsaitwal** [comment]
> What do you think if I create a new Reddit post (or just add a comment in the current post?) asking for people to suggest and vote names?

Nice idea, go ahead. 👍🏻

**Comment by fzyzcjy** [comment]
I am worried: I have already create one post yesterday about this lib. If I create one today, will people dislike this?

**Comment by shreyashsaitwal** [comment]
Given how great this project is, I don't think anyone would dislike it. But because the current post has already got some good engagement, IMO editing it (or adding a comment in it) would be a good idea.

**Comment by fzyzcjy** [comment]
Thanks. Let me edit it.

**Comment by fzyzcjy** [comment]
I have edited it. Did you receive any kinds of notifications?

**Comment by shreyashsaitwal** [comment]
> Did you receive any kinds of notifications?

Nope, I didn't (and I've no idea if other people who commented would receive it. I don't use Reddit much). But after I refreshed the page I saw the edit.

**Comment by coderedart** [comment]
I think any name is fine that is short. something like `flut-rs` or anything like that. people won't get reddit notifications when you edit post :) If you want to ask the community, ask in rust and rust-community discords, where people can reply immediately.

**Comment by coderedart** [comment]
anyone who searches for a "flutter rust" is going to find it anyway because of the description or ReadMe. name doesn't really need to be too descriptive.

**Comment by shreyashsaitwal** [comment]
Okay, this just struck my mind rn. What if we called it `ferb`? Inspired from **F**lutter **R**ust **B**ridge (`frb`).

**Comment by fzyzcjy** [comment]
@shreyashsaitwal That sounds like an animation https://en.wikipedia.org/wiki/Ferb_Fletcher

**Comment by shreyashsaitwal** [comment]
> That sounds like an animation https://en.wikipedia.org/wiki/Ferb_Fletcher

Oh, that might cause copyright overhead I guess. Thanks for pointing it out.

**Comment by fzyzcjy** [comment]
@shreyashsaitwal You are welcome!

**Comment by shreyashsaitwal** [comment]
Sorry, I should probably have created a new issue.

**Comment by fzyzcjy** [comment]
No problem, I have renamed it

**Comment by fzyzcjy** [comment]
I come up with a few names. What do you guys think?

* `fridge`: Flutter Rust brIDGE
* `flat`: FLutter And rusT
* `rust_dart`

**Comment by fzyzcjy** [comment]
Maybe I choose `fridge`?

**Comment by shreyashsaitwal** [comment]
Yes, `fridge` sounds nice.

**Comment by fzyzcjy** [comment]
Oops, `fridge rust` means your fridge has some rust on it when you google it... Let me think twice

**Comment by coderedart** [comment]
I think this is getting way more efffort put in, than it is needed. half the rust libraries get drowned in rust (iron oxide) related search results, and the rest get drowned in the rust (the game) results. fridge should be fine. or flust/flut . rust_dart seems like a reasonable option too. if this turns out great, people will not really care about the name.

**Comment by fzyzcjy** [comment]
Well let me think twice... Some of my friends think the current name is better. So let me focus on the code itself first

**Comment by shreyashsaitwal** [comment]
I completely agree with @coderedart. You're thinking way too much about this. Every new project takes some time to show up in search engine results. And how would you get any relevant result even before renaming the project?!

Also, if you don't want to change the name, it's completely fine. No one's forcing you to do so. At the end of the day, what matters is the project itself, not it's name.

**Comment by mjablecnik** [comment]
And what about name `dart_rust_bridge`?

**Comment by shreyashsaitwal** [comment]
Again, it's too big.

**Comment by mjablecnik** [comment]
No it is not too big name I think..

**Comment by coderedart** [comment]
Very few popular crates have 3 words, and I think having a _ is always confusing. I would still recommend a single word name or 5 letters like flust or flur or drust or darust or flutrust or rufler etc..

**Comment by mjablecnik** [comment]
It is not about name. It is about functionality what depends about how package will be popular or not..
In https://pub.dev is a lot of packages with quite same long names and `_`:
![image](https://user-images.githubusercontent.com/2686906/136339734-e8cec0bb-410a-4b98-963a-1565de957626.png)

**Comment by coderedart** [comment]
Yeah. That was what I said long ago in this thread. But this thread literally is about naming, so I recommended short names, and it's easier to remember one word than 2.

**Comment by sachaarbonel** [comment]
Feel free to take [dodge](https://github.com/sachaarbonel/dodge), when I was searching for a name, looking for "rust dart" I was finding images of Dodge cars. It's nice because of dodge in a sense of "avoid doing bindings by hand"

**Comment by fzyzcjy** [comment, marked as answer]
Let's just firstly keep flutter_rust_bridge ;)
